TE-NS ubuntu migration 22.04

priyadarshitathagat opened the pull request:
Signed-off-by: priyadarshitathagat <tpriyadarshi@vmware.com>

priyadarshitathagat commented:
The comments will be added post testing and validation. Would add comments as to why each change is required.

aravindhank11 commented on te_dp/Makefile:
```diff
@@ -39,20 +39,20 @@ BINDIR = bin
 
 dp: $(DP_OBJ)
-	@$(CC) $(CFLAGS) -o $(DP_BIN) $^ $(DP_FLAGS)
+	@$(CC) $(CFLAGS) -o $(DP_BIN) $^ $(DP_FLAGS) -no-pie
```
Could you please share the reasoning and need for `-no-pie` flag?

priyadarshitathagat replied:
https://stackoverflow.com/questions/73275183/gcc-assembly-x86-relocation-warning-creating-dt-textrel-in-a-pie 
it was due to this issue

aravindhank11 commented on te/tedp_docker/Dockerfile:
```diff
@@ -53,7 +53,7 @@ RUN apt update && \
 # python related requirements
 COPY te/tedp_docker/requirements.txt /tmp/
-RUN pip3 install --upgrade "pip < 21.0"
+RUN pip3 install --upgrade "pip == 22.3.1"
```
Should we pin pip to a version? Again this was done to avoid package breakage in pip specific to ubuntu 16 getting EOL. Is it needed?

priyadarshitathagat replied:
ACK.

aravindhank11 commented on te/te_docker/setup_postgres.sh:
```diff
@@ -32,7 +32,7 @@
 apt-get install -y wget
 wget --no-check-certificate --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
-sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ xenial-pgdg main" > /etc/apt/sources.list.d/PostgreSQL.list'
+sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ jammy-pgdg main" > /etc/apt/sources.list.d/PostgreSQL.list'
```
We might not need to get postgres from custom postgres repo. We could get it from ubuntu repo. Could we please consider removing this ?

aravindhank11 replied:
Also move postgres to the latest version available. IG Ubuntu support psql 14

aravindhank11 commented on te/te_docker/Dockerfile:
```diff
@@ -75,12 +75,14 @@ missingok \n\
 
 COPY te_docker/requirements.txt /tmp/
-RUN pip3 install --upgrade "pip < 21.0"
+RUN apt install libpq-dev -y
```
Why is dev package needed in the image?

priyadarshitathagat replied:
https://stackoverflow.com/questions/35104097/how-to-install-psycopg2-with-pg-config-error

aravindhank11 commented on te/TE.py:
```diff
@@ -1333,14 +1353,17 @@ def clean_api(self, jsonContent):
                 return self.__failure("No connected te dps to disconnect")
 
-            client = ParallelSSHClient(te_dp_hosts.keys(), host_config=te_dp_hosts, timeout = 240)
+            client = ParallelSSHClient(list(te_dp_hosts.keys()), host_config=tedp_host_config, timeout = 240)
```
Is this change relevant to OS upgrade? If not, please consider raising a different one. 

If it is, please share details onto what is this change needed?

priyadarshitathagat replied:
Yes, these are all related to 22.04 migration - since we intend on using python3.10 and a lot of modules/api has changed.

priyadarshitathagat commented:
@aravindhank11 - we are in the process of testing and we are still in the process of fixing a lot of package dependency and modules changes that has come in due to python3.10.  I have compiled all the notes regarding each and every change, will update the PR with my comments soon as mentioned earlier. For now the PR is not open for review. 

Will tag everyone once all the changes as completed and open for review - I will have to minimise and trim a lot of changes that were added initially to get the compilation and docker going for testing.

priyadarshitathagat commented on te/tedp_docker/tedp_bin_builder.dockerfile:
```diff
@@ -50,12 +50,12 @@ RUN chmod 755 /tmp/setup.sh
 ADD te_dp/Makefile $WORKDR
 ADD te_dp/src $WORKDR/src
-RUN cd $WORKDR && make all
+RUN cd $WORKDR && make clean && make all
```
ack - this was during my compilation - will remove

priyadarshitathagat commented on te/tedp_docker/setup.sh:
```diff
@@ -70,16 +71,16 @@ cd /tmp/
 rm -rf openssl*
 
 #LIBCURL
-cd curl-7.67.0
```
I will check with @vipinpr5991 also one and get back, while using curl-7.67 I was hitting an issue during compilation 
while i also tried curl-7.87 - but the the libcurl api has changed and stopped supporting a few, it would require a lot of C code changes to get this done. The intent was minimum penetration/changes in C code.

priyadarshitathagat commented:
Created a branch to track every internal JIRA reg ubuntu migration bugs/fixes - 
Branch -> `tens-ubuntu-22.04-migration` 

Will consider addressing all the comments from here. Closing this one.
